src/api/migraphx.py

```diff
@@ -176,9 +181,7 @@ def onnx_options(h):
     h.constructor('create')
     h.method(
         'set_input_parameter_shape',
-        api.params(name='const char*',
-                   dims='const size_t *',
```

Comment:
Same as above, the reason we change to this API is to process the case of dims.datat() is nullptr.